src/core/thread/mle_router.cpp

```diff
@@ -2502,6 +2511,16 @@ void MleRouter::SetLeaderWeight(uint8_t aWeight)
     mLeaderWeight = aWeight;
 }
 
+uint32_t MleRouter::GetLeaderPartitionId(void) const
+{
+	return mLeaderPartitionId;
+}
+
+void MleRouter::SetLeaderPartitionId(uint32_t aPartitionId)
+{
+	mLeaderPartitionId = aPartitionId;
```

Comment:
Actually here what we want to return is the local leader Partition id to be used when Leader creates a new thread partition, not the current partition id from leader data tlv. And we can obtain the current real partition id via "leaderdata" cli. Maybe that's not conflict with local leader partition id meaning, is similar with "leaderweight" cil.